src/main/java/de/unistuttgart/informatik/fius/icge/territory/WorldObject.java

```diff
@@ -13,39 +13,30 @@
 
 public class WorldObject implements Comparable<WorldObject> {
 
-    public final EntityType type;
+    public final EntityState state;
     public final int column;
     public final int row;
     public final Direction direction;
     public final float z;
 
-    public WorldObject(EntityType type, int column, int row, float z, Direction direction) {
+    public WorldObject(EntityState state, int column, int row, float z, Direction direction) {
         this.column = column;
         this.row = row;
-        this.type = type;
+        this.state = state;
         this.z = z;
         this.direction = direction;
     }
 
-    public WorldObject(EntityType type, int column, int row, Direction direction) {
-        this(type, column, row, 0, direction);
+    public WorldObject(EntityState state, int column, int row, Direction direction) {
+        this(state, column, row, 0, direction);
     }
 
-    public WorldObject(EntityType type, int column, int row, float z) {
-        this(type, column, row, z, Direction.EAST);
+    public WorldObject(EntityState state, int column, int row, float z) {
+        this(state, column, row, z, Direction.EAST);
     }
 
-    public WorldObject(EntityType type, int column, int row) {
-        this(type, column, row, Direction.EAST);
-    }
-
-    @Override
-    public boolean equals(Object other) {

```

Comment:
Why remove the equals method?

Reply:
It didn't check for the equality of the state and I think the `equals` method should be deep if it exists.